Code written for LinkIt

20kvnk opened the pull request:
The device detects upto a range of 22-24m comfortably even with objects in the way. Sitting in a bus doesn't have much effect on the range.
Range is negligibly more than Edison, like a meter max. So, both have like the same range.

Since scan time is 3 seconds, scans nearby devices comfortably. An issue is that the number of devices being detected after one scan is not very consistent.Taking the MAC addresses after 2-3 scans gives a more accurate number of BT devices in the area.
Also, on the website it said that a scan can detect at max. 10 devices. 
(An Idea, Not Implemented, Subject to discussion) So, thought of creating sets to store the data provided in two three scans and then sending this set to the sd card. 

Another issue is that, let's say there is a person walking on the pavement and using bluetooth, like on his phone or bluetooth headest. The device cannot distinguish between a vehicle and that person.

A problem in the code is that even though data is to be sent to the mbt.txt file on sd card,
an error props up, saying that it can't even open such a file.
On the serial monitor, data is being shown normally. 
Opening the .txt file is not being implemented. Data is not being sent to sd card. Couldn't find a way to resolve the problem.

20kvnk committed:
Code written for LinkIt


The device detects upto a range of 22-24m comfortably even with objects in the way. Sitting in a bus doesn't have much effect on the range.
Range is negligibly more than Edison, like a meter max. So, both have like the same range.

Since scan time is 3 seconds, scans nearby devices comfortably. An issue is that the number of devices being detected after one scan is not very consistent.Taking the MAC addresses after 2-3 scans gives a more accurate number of BT devices in the area.
Also, on the website it said that a scan can detect at max. 10 devices. 
(An Idea, Not Implemented, Subject to discussion) So, thought of creating sets to store the data provided in two three scans and then sending this set to the sd card. 

Another issue is that, let's say there is a person walking on the pavement and using bluetooth, like on his phone or bluetooth headest. The device cannot distinguish between a vehicle and that person.

A problem in the code is that even though data is to be sent to the mbt.txt file on sd card,
an error props up, saying that it can't even open such a file.
On the serial monitor, data is being shown normally. 
Opening the .txt file is not being implementd. Data is not being sent to sd card. Couldn't find a way to resolve the problem.

kunalgrover05 commented on LinkitOne/bluetooth.ino:
```diff
@@ -0,0 +1,62 @@
+
+
+LFile mbt;
+
```
Please indent your code properly.

kunalgrover05 commented on LinkitOne/bluetooth.ino:
```diff
@@ -0,0 +1,62 @@
+                    
+  }
+mbt.close();
+delay(1);
```
Do we need a delay of 1 second here?

20kvnk replied:
The delay is just to make sure that if there is any problem, the microcontroller and the data on it doesn't get affected. It's just for failsafe purposes. Nothing else.

kunalgrover05 replied:
Then please test the minimum delay needed, 1 sec might be an overkill and we may lose essential data.

kunalgrover05 commented on LinkitOne/bluetooth.ino:
```diff
@@ -0,0 +1,62 @@
+#include <LBT.h>
```
Can you please on top write a comment stating what all this file does?

kunalgrover05 replied:
Can you please on top write a comment stating what all this file does?

kunalgrover05 commented:
According to your comment, SD doesn't work yet? Can you write a comment on your file stating that?
Also add a new folder called Experiments, create a new text file with today's date, and add things from your comment above in 2 sections, one for experiment results and second for issues in the current model

20kvnk committed:
Update bluetooth.ino

20kvnk committed:
SD Card issue

The problem in the code is that the file, mbt.txt which was to open to store scanned data, couldn't be opened.
An error message was returned.

20kvnk commented:
Comment on the file for sd card issue.....in the actual code?

20kvnk committed:
Update bluetooth.ino

20kvnk committed:
Experiment Results- 11 Apr

Contains the results of experiments done on the LinkIt device as of 11 Apr 2016

20kvnk committed:
Rename Experiments/Updates_11Apr to LinkIt-Experiments/Updates_11Apr

kunalgrover05 commented on LinkitOne/bluetooth.ino:
```diff
@@ -0,0 +1,64 @@
+""" Declaring a file to be stored on the sd card """
+LFile mbt;
+
+void setup() {
```
Please indent your code. Follow Arduino's default coding style and indent your code.

kunalgrover05 commented on LinkIt-Experiments/Updates_11Apr:
```diff
@@ -0,0 +1,4 @@
+Results
```
I meant you need to fill this file with the results that you had posted with your PR about testing(preferably as bullets). It will keep a documentation about what tests we did and what results we got.

Same for issues below, put issues that we faced here, for eg the readings we got weren't consistent.

20kvnk replied:
I finished the Updates text.

20kvnk committed:
Update Updates_11Apr

20kvnk committed:
Update bluetooth.ino

20kvnk commented:
Code Indented

kunalgrover05 commented on LinkitOne/bluetooth.ino:
```diff
@@ -0,0 +1,63 @@
+
+    mbt = LSD.open("mac.txt", FILE_WRITE);
+    if (!mbt)
+
```
Follow same indenting style everywhere, one place has 
loop() {
while other has
if() 
{

Best way to do it is, always auto-indent using Tools->Auto Format

20kvnk committed:
Update bluetooth.ino
